Marlin/stepper_indirection.h

```diff
@@ -418,38 +418,43 @@
 #if ENABLED(SWITCHING_EXTRUDER)
   #if EXTRUDERS == 2
     #define E_STEP_WRITE(v) E0_STEP_WRITE(v)
-    #define NORM_E_DIR() E0_DIR_WRITE(current_block->active_extruder ?  INVERT_E0_DIR : !INVERT_E0_DIR)
-    #define  REV_E_DIR() E0_DIR_WRITE(current_block->active_extruder ? !INVERT_E0_DIR :  INVERT_E0_DIR)
+    #define NORM_E_DIR() E0_DIR_WRITE(TOOL_E_INDEX ?  INVERT_E0_DIR : !INVERT_E0_DIR)
+    #define  REV_E_DIR() E0_DIR_WRITE(TOOL_E_INDEX ? !INVERT_E0_DIR :  INVERT_E0_DIR)
```

Comment:
While this will work, it gives the false impression that `TOOL_E_INDEX` might be something other than `current_block->active_extruder`, when it certainly will never be anything other than `current_block->active_extruder`.